### PR TITLE
manager.py broadcast process states

### PR DESCRIPTION
**Pull request by pd0wm**

### CODE REVIEW

File: selfdrive/controls/lib/events.py [hunk 1]
```diff
@@ -615,6 +615,11 @@ def wrong_car_mode_alert(CP: car.CarParams, sm: messaging.SubMaster, metric: boo
                               audible_alert=AudibleAlert.chimeDisengage),
   },
 
+  EventName.processNotRunning: {
+    ET.NO_ENTRY: NoEntryAlert("Not All Processes Are Running",
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/events.py:619]: what about something like system malfunction? I think this has too much information for a user and can cause confusion. what's a process? will they start up in a minute if I try to reenagage?

Comment (pd0wm) [reply selfdrive/controls/lib/events.py:619]: Something like: "System Malfunction. Reboot Device"?

**Comment by pd0wm** [comment]
We also need a way for this check to not trigger during tests in CI. We can do two things:
 - Change all tests to also broadcast a managerState packet
 - Only trigger the check if we have seen at least 1 managerState packet.

I'd like to do the latter.

File: selfdrive/controls/controlsd.py [hunk 2]
```diff
@@ -166,6 +167,11 @@ def update_events(self, CS):
     if self.sm['thermal'].memUsedPercent > 90:
       self.events.add(EventName.lowMemory)
 
+    # Check if all manager processes are running
+    not_running = set(p.name for p in self.sm['managerState'].processes if not p.running)
+    if self.sm.rcv_frame['managerState'] and (not_running - IGNORE_PROCESSES):
```
Comment (adeebshihadeh) [inline selfdrive/controls/controlsd.py:172]: hmm, why don't we add a timeout to rcv_frame? it's possible for this check to not work if the packet is never published

Comment (adeebshihadeh) [reply selfdrive/controls/controlsd.py:172]: guess the alive check will catch this

**Pull request merged**